src/test/rpc/Simulate_test.cpp

```diff
@@ -0,0 +1,989 @@
+
+                BEAST_EXPECT(result[jss::engine_result] == "tesSUCCESS");
+                BEAST_EXPECT(result[jss::engine_result_code] == 0);
+                BEAST_EXPECT(
+                    result[jss::engine_result_message] ==
+                    "The simulated transaction would have been applied.");
+
+                if (BEAST_EXPECT(
+                        result.isMember(jss::meta) ||
+                        result.isMember(jss::meta_blob)))
+                {
+                    Json::Value metadata;
+                    if (result.isMember(jss::meta_blob))
+                    {
+                        auto unHexed =
+                            strUnHex(result[jss::meta_blob].asString());
+                        SerialIter sitTrans(makeSlice(*unHexed));
+                        metadata = STObject(std::ref(sitTrans), sfGeneric)
+                                       .getJson(JsonOptions::none);
+                    }
+                    else
+                    {
+                        metadata = result[jss::meta];
+                    }
+                    if (BEAST_EXPECT(
+                            metadata.isMember(sfAffectedNodes.jsonName)))
+                    {
+                        BEAST_EXPECT(metadata[sfAffectedNodes].size() == 1);
+                        auto node = metadata[sfAffectedNodes][0u];
+                        if (BEAST_EXPECT(
+                                node.isMember(sfModifiedNode.jsonName)))
+                        {
+                            auto modifiedNode = node[sfModifiedNode];
+                            BEAST_EXPECT(
+                                modifiedNode[sfLedgerEntryType] ==
+                                "AccountRoot");
+                            auto finalFields = modifiedNode[sfFinalFields];
+                            BEAST_EXPECT(finalFields[sfDomain] == newDomain);
+                        }
+                    }
+                    BEAST_EXPECT(metadata[sfTransactionIndex] == 1);
+                    BEAST_EXPECT(metadata[sfTransactionResult] == "tesSUCCESS");
+                }
+            };
+
+            Json::Value tx;
+
+            tx[jss::Account] = alice.human();
+            tx[jss::TransactionType] = jss::AccountSet;
+            tx[sfDomain] = newDomain;
+            tx[sfSigners] = Json::arrayValue;
+            {
+                Json::Value signer;
+                signer[jss::Account] = becky.human();
+                signer[jss::SigningPubKey] = strHex(becky.pk().slice());
+                Json::Value signerOuter;
+                signerOuter[sfSigner] = signer;
+                tx[sfSigners].append(signerOuter);
+            }
+
+            // test with autofill
+            testTx(env, tx, validateOutput, false);
+
+            tx[sfSigningPubKey] = "";
+            tx[sfTxnSignature] = "";
+            tx[sfSequence] = 5;
+            tx[sfFee] = "20";  // also tests a non-base fee
```

Comment:
"20" - > please change to calculated value (baseFee * 2 or something)

Reply:
Fixed in [f1ac9b8](https://github.com/XRPLF/rippled/pull/5069/commits/f1ac9b8dc444105d8227afbb6089c9486c351500)